src/main/scala/SemanticVersion.scala

```diff
@@ -1,41 +1,28 @@
 package compliant
 
-class SemanticVersion(val versionString: String) {
-  val trimmedVersionString = versionString.trim.replaceAll("\\s", "")
-  val splitByPeriods = trimmedVersionString.split("\\.")
-  val major = splitByPeriods(0)
-  val minor = splitByPeriods(1)
-  val patch = splitByPeriods.slice(2,splitByPeriods.length).mkString(".").split("\\+|\\-")(0)
+import scala.util.matching.Regex
 
-  def fullVersionString: String = {
-    this.trimmedVersionString
-  }
+object SemanticVersion {
 
-  def majorVersion : String = {
-    this.major
-  }
+  val re: Regex = """(\d+)\.(\d+)\.(\d+)(-([^+]+))?(\+(.+))?""".r
 
-  def minorVersion : String = {
-    this.minor
-  }
+}
 
-  def patchVersion : String = {
-    this.patch
-  }
+class SemanticVersion(val versionString: String) {
 
-  def versionStringWithoutBuildMetadata : String = {
-    this.trimmedVersionString.split("\\+").head
-  }
+  val fullVersionString: String = versionString.trim.replaceAll("\\s", "")
 
-  def prerelease : String = {
-    val splitByDashes = this.trimmedVersionString.split("\\-")
-    splitByDashes.slice(1, splitByDashes.length).mkString("-").split("\\+").head
+  val (majorVersion, minorVersion, patchVersion, prerelease, buildMetadata) = {
+    fullVersionString match {
+      case SemanticVersion.re(major, minor, patch, _, pre, _, meta) =>
+        (major, minor, patch, Option(pre).getOrElse(""), Option(meta).getOrElse(""))
```

Comment:
It may be advantageous to leave `prerelease` and `buildMetadata` as `Option` types.